src/test/ui/consts/const-loop.rs

```diff
@@ -0,0 +1,58 @@
+const _: i32 = {
+    let mut x = 0;
+
+    while x < 4 {
+        //~^ ERROR constant contains unimplemented expression type
+        //~| ERROR constant contains unimplemented expression type
+        x += 1;
+    }
+
+    while x < 8 {
```

Comment:
Does the checker stop before reaching here?

Reply:
Yes. The current one does. The new one gives much better errors on code like this but enabling it is blocked on #65949.

For background, I was using this file as a test case while working on improving diagnostics for the new checker, and figured it could be merged upstream separately.